src/PrestaShopBundle/Command/AppendConfigurationFileHooksListCommand.php

```diff
@@ -0,0 +1,229 @@
+
+        $io = new SymfonyStyle($input, $output);
+
+        if (!in_array($container->getParameter('kernel.environment'), ['dev', 'test'])) {
+            $io->warning('Dev or test environment is required to fully list all the hooks');
+
+            return;
+        }
+
+        $hookNames = $this->getHookNames();
+
+        try {
+            $addedHooks = $this->appendHooksInConfigurationFile($hookNames);
+        } catch (Exception $e) {
+            $io->error($e->getMessage());
+        }
+
+        if (!empty($addedHooks)) {
+            $io->title('Hooks added to configuration file');
+            $io->note(sprintf('Total hooks added: %s', count($addedHooks)));
+            $io->listing($addedHooks);
+
+            return;
+        }
+
+        $io->note('No new hooks have been added to configuration file');
+    }
+
+
+    /**
+     * Initialize PrestaShop Context
+     */
+    private function initContext()
+    {
+        require_once $this->getContainer()->get('kernel')->getRootDir() . '/../config/config.inc.php';
+        /** @var LegacyContext $legacyContext */
+        $legacyContext = $this->getContainer()->get('prestashop.adapter.legacy.context');
+        //We need to have an employee or the listing hooks don't work
+        //see LegacyHookSubscriber
+        if (!$legacyContext->getContext()->employee) {
+            //Even a non existing employee is fine
+            $legacyContext->getContext()->employee = new Employee(42);
+        }
+    }
+
+    /**
+     * Gets all hooks names which need to be appended.
+     *
+     * @return string[]
+     */
+    private function getHookNames()
+    {
+        $container = $this->getContainer();
+
+        $gridServiceIds = $container->getParameter('prestashop.core.grid.definition.service_ids');
+        $optionsFormHookNames = $container->getParameter('prestashop.hook.option_form_hook_names');
+        $identifiableObjectFormTypes = $container->getParameter('prestashop.core.form.identifiable_object.form_types');
+
+        $gridDefinitionHooksProvider = $container->get(
+            'prestashop.core.hook.provider.grid_definition_hook_by_service_ids_provider'
+        );
+
+        $identifiableObjectFormTypeProvider = $container->get(
+            'prestashop.core.hook.provider.identifiable_object_hook_by_form_type_provider'
+        );
+
+        $gridDefinitionHookNames = $gridDefinitionHooksProvider->getHookNames($gridServiceIds);
+
+        $identifiableObjectHookNames = $identifiableObjectFormTypeProvider->getHookNames($identifiableObjectFormTypes);
+
+        return array_merge(
+            $identifiableObjectHookNames,
+            $optionsFormHookNames,
+            $gridDefinitionHookNames
+        );
+    }
+
+    /**
+     * Appends given hooks in the configuration file.
+     *
+     * @param array $newHookNames
+     *
+     * @return array
+     *
+     * @throws Exception
+     */
+    private function appendHooksInConfigurationFile(array $newHookNames)
```

Comment:
it uses hardcoded path `install-dev/data/xml/` which will only exist in our development environment - that's the reason I don't want to move it to any service.

I only see that I can split here some kind of xml content modification service but I really doubt about it :thinking: wdyt of how it should look like?

Reply:
> it uses hardcoded path install-dev/data/xml/ which will only exist in our development environment - that's the reason I don't want to move it to any service.

I don't see why this could be a reason not to extract logic in separate service. @matks wdyt?